### ISSUE TITLE
Support for upstream uwsgi protocol

### ISSUE THREAD

**Issue by mattrobenolt** [envoyproxy/envoy#15030]
It'd be nice to be able to support the uwsgi binary protocol for an upstream cluster when speaking to a service that uses this.

Fundamentally this is a lighter weight protocol when you're running a service behind uwsgi already, you can skip another layer of http processing and speak the native uwsgi protocol.

Prior art: nginx supports a `uwsgi_pass` directive: https://nginx.org/en/docs/http/ngx_http_uwsgi_module.html
Ref: https://uwsgi-docs.readthedocs.io/en/latest/Protocol.html

**Comment by noxiouz** [comment]
I am new to Envoy and I am looking for an opportunity to contribute to it. That one looks like a good feature to dive into Envoy. So the Envoy team is ok with that, I am happy to implement that module.

**Comment by mattklein123** [comment]
I think with the new extension upstream support this should be pretty easy, but it would be good to have a design doc. @alyssawilk wdyt?

**Comment by alyssawilk** [comment]
Yeah, I think it should be pretty easy to plug it in.  I think to me the main question is if there's enough interest to land upstream, and if any of the @envoyproxy/maintainers want to sign up to co-own.  

(https://github.com/envoyproxy/envoy/blob/main/EXTENSION_POLICY.md#adding-new-extensions)

**Comment by phlax** [comment]
>  if there's enough interest to land upstream

throwing in my 2p - this would be hugely beneficial to python communities and i think in terms of users would likely get a lot of adoption

**Comment by phlax** [comment]
bump

**Comment by gnalsa** [comment]
Yep, having nginx->uwsgi is common pattern. If envoy has native support for wsgi protocol it could help eliminate nginx and reduce complexity of deployment.

**Comment by viginti23** [comment]
Still would love to see that feature :)

**Comment by kdelee** [comment]
👀
